core/modules/book/book.module

```diff
@@ -1243,3 +1249,42 @@ function book_menu_subtree_data($link) {
 
   return $tree[$cid];
 }
+
+/**
+ * Invalidates cached data relating to book_cache.
+ *
+ * @param $cid
+ *   (optional) Cache ID of the record to clear from the private update module
+ *   cache. If empty, all records will be cleared from the table except fetch
+ *   tasks. Defaults to NULL.
+ * @param $wildcard
+ *   (optional) If TRUE, cache IDs starting with $cid are deleted in addition to
+ *   the exact cache ID specified by $cid. Defaults to FALSE.
+ */
+function _book_cache_clear($cid = NULL, $wildcard = FALSE) {
+  if (empty($cid)) {
+    cache_clear_all('*', 'cache_book', TRUE);
+  }
+  else {
+    cache_clear_all($cid, 'cache_book', $wildcard);
+  }
+}
+
+/**
+ * Implements hook_flush_caches().
+ */
+function book_flush_caches() {
+  _book_cache_clear();
+  return array();
```

Comment:
This should return the cache bin name, not clear it immediately. i.e. `return array('book');`